starter_code/public/javascript/index.js

```diff
@@ -1,23 +1,81 @@
-const charactersAPI = new APIHandler("http://localhost:8000")
+const charactersAPI = new APIHandler("http://localhost:8000");
 
-$(document).ready( () => {
-  document.getElementById('fetch-all').onclick = function(){
+window.onload = () => {
+  document.getElementById("fetch-all").onclick = function() {
+    charactersAPI
+      .getFullList()
+      .then(res => {
+        $(".characters-container").empty();
+        res.forEach(element => {
+          $(".characters-container").append(
+            `<div class="character-info">
+              <div class="id">Id : ${element.id}</div>
+              <div class="name">Name : ${element.name}</div>
+              <div class="occupation">Occupation : ${element.occupation}</div>
+              <div class="debt">Debt : ${element.debt}</div>
+              <div class="weapon">Weapon : ${element.weapon}</div>
+            </div>`
+          );
+        });
+      })
+      .catch(err => console.error(err));
+  };
 
-  }
-  
-  document.getElementById('fetch-one').onclick = function(){
-    
-  }
-  
-  document.getElementById('delete-one').onclick = function(){
-        
-  }
-  
-  document.getElementById('edit-character-form').onsubmit = function(){
-            
-  }
-  
-  document.getElementById('new-character-form').onsubmit = function(){
-                
-  }
-})
+  document.getElementById("fetch-one").onclick = function() {
+    let characterId = $("#charId").val();
+    console.log(characterId);
+    charactersAPI
+      .getOneRegister(characterId)
+      .then(res => {
+        $(".characters-container").empty();
+        $(".characters-container").append(
+          `<div class="character-info">
+            <div class="id">Id : ${res.id}</div>
+            <div class="name">Name : ${res.name}</div>
+            <div class="occupation">Occupation : ${res.occupation}</div>
+            <div class="debt">Debt : ${res.debt}</div>
+            <div class="weapon">Weapon : ${res.weapon}</div>
+          </div>`
+        );
+      })
+      .catch(err => console.error(err));
+  };
+
+  document.getElementById("delete-one").onclick = function() {
+    event.preventDefault();
```

Comment:
You need to declare the event object in the function sugnature:

` document.getElementById("delete-one").onclick = function(event) {  ... ...`